hibike/devices/YogiBear/LED.cpp

```diff
@@ -2,41 +2,61 @@
 #include "pindefs.h"
 #include "encoder.h"
 #include "current_limit.h"
-
+#include "motor.h"
 #include "Arduino.h"
 
+bool redOn = false;
+bool yellowOn = false;
+bool greenOn = false;
+
 //place all LED control functions in here
 void ctrl_LEDs() {
-	ctrl_RED();
-	ctrl_YELLOW();
-	ctrl_GREEN();
-}
-
-//decides when the red LED is on
-void ctrl_RED() {
+	//control red LED state
 	if(readVel() < 0) {
+		redOn = true;
```

Comment:
Uh oh i messed up on one of my previous comments. It actually makes sense to divide the ctrl_GREEN/RED/YELLOW functions, cause i missed the if statements being different. Sorry! x_x Please revert.

However, the digitalWrite factoring out is good. 
I wouldn't use globa booleans though. Just instantiate them in the function.